### PR TITLE
Primitives refactor

### PR DESCRIPTION
**Pull request by Degubi**
Refactor & Clean up the Primitives class to use array lookup instead of HashMap lookup. There's no need to use a HashMap for 9 elements.
Benchmark code: https://pastebin.com/cG8w2pyy
Benchmark output: https://pastebin.com/dCKfacCy

### CODE REVIEW

File: gson/src/main/java/com/google/gson/internal/Primitives.java [hunk 1]
```diff
@@ -29,47 +25,34 @@
...
+	return -1;
   }
-
+  
   /**
    * Returns true if this type is a primitive.
    */
   public static boolean isPrimitive(Type type) {
-    return PRIMITIVE_TO_WRAPPER_TYPE.containsKey(type);
+    return indexOf(type, 0) != -1;
```
Comment (JakeWharton) [inline gson/src/main/java/com/google/gson/internal/Primitives.java]: Isn't this just `type instanceof Class<?> && ((Class<?>) type).isPrimitive()`?

**Comment by Degubi** [comment]
I was thinking about that, I will update to use normal if-s instead of array lookup.

**Comment by Degubi** [comment]
I think this looks good now.

**Comment by JakeWharton** [comment]
Reformatted, squashed, and merged as 477b3f77e159d15531fd30f41688d4b732a8b042. Thanks!

**Pull request closed without merging**